floodsub.go

```diff
@@ -319,6 +365,32 @@ func msgID(pmsg *pb.Message) string {
 	return string(pmsg.GetFrom()) + string(pmsg.GetSeqno())
 }
 
+// validate is called in a goroutine and calls the validate functions of all subs with msg as parameter.
+func (p *PubSub) validate(subs []*Subscription, msg *Message) bool {
+	for _, sub := range subs {
+		ctx, cancel := context.WithTimeout(p.ctx, sub.validateTimeout)
+		defer cancel()
+
+		result := make(chan bool)
+		go func(sub *Subscription) {
+			result <- sub.validate == nil || sub.validate(ctx, msg)
+		}(sub)
+
+		select {
```

Comment:
this select should probably go in a separate loop, after this one. That way we do all the validation in parallel

Reply:
Yep, good catch!